doc/syntax_error.md

```diff
@@ -0,0 +1,66 @@
+# SyntaxError
+
```

Comment:
Should it say here that it's only supported in node-api version 9 and higher?

Reply:
That might be a gap though ?  I go to try to use an API and it's not there?  Either way agree that we don't need to be block this PR on it, but might be worth thinking about wether adding that info makes sense.